misc-scripts/xref_mapping/XrefParser/JGI_Parser.pm

```diff
@@ -76,89 +74,60 @@ sub run {
   #  DEDDDL*
   #
 
-
-
   my @xrefs;
 
-  local $/ = "\n>";
-
   my $file_io = $self->get_filehandle($file);
 
   if ( !defined $file_io ) {
-    print STDERR "ERROR: Could not open $file\n";
-    return 1;    # 1 is an error
+    croak "ERROR: Could not open $file\n";
   }
 
-  while ( $_ = $file_io->getline() ) {
+  IO::Handle->input_record_separator( "\n>");
+  
+  while ( my $record = $file_io->getline() ) {
 
-    next if (/^File:/);   # skip header
+    next if ($record =~ /^File:/);   # skip header
 
     my $xref;
 
-    my ($header, $sequence) = $_ =~ /^>?(.+?)\n([^>]*)/s or warn("Can't parse FASTA entry: $_\n"); 
+    my ($header, $sequence) = $record =~ /^>?(.+?)\n([^>]*)/s or croak("Can't parse FASTA entry: $record\n");
+    my $accession = $header;
 
-      my @attr = split/\|/, $header ; 
-      my $acession = $header ;
-    
     # split header in different ways according to source name : 
-    my ( $version,$label  ) ;      
-    if ($source_name=~m/cint_jgi_v1/) { 
+
+    if ($source_name=~m/cint_jgi_v1/) {
       # header format is  >ci0100146277
-      # we want  146277
-      ($acession = $header)  =~s/\w{6}//;  
-      $version = "JGI 1.0" ;  
-    
-    } elsif ($source_name=~m/cint_aniseed_.*v1/) { 
+      # JGI 1.0
+      # we want accession 146277 from above
+      ($accession = $header) =~s/\w{6}//;
+
+    } elsif ($source_name=~m/cint_aniseed_.*v1/) {
       # header format is  >ci0100146277, we want this 
-      $version = "JGI 1.0" ;  
-
-
-    } elsif ($source_name=~m/cint_jgi_v2/) { 
-      $acession = $attr[2]  ; 
-      $version = "JGI 2.0" ;  
-      $label = $attr[3] ; 
-    } elsif ($source_name=~m/cint_aniseed_.*v2/) { 
-      my $aniseed_prefix = "ci0200" ; 
-      $acession = $aniseed_prefix . $attr[2]  ; 
-      $version = "JGI 2.0" ;  
-
-    }else { 
-      print STDERR "WARNING : The source-name specified ($source_name) in the populate_metatable.sql file is\n" .
-        "WARNING : not matching the differnt cases specified in JGI_Parser.pm - plese\n".
-          "WARNING : edit the parser \n" ; 
-      return 1;
-    } 
-    #print "ACCESSION $acession\n" ;  
+      # JGI 1.0
+
+    } else {
+      croak "WARNING : The source-name specified ($source_name) in the populate_metatable.sql file is\n" .
```

Comment:
I find it's a little bit misleading to emit a warning when dying: I'd rather say: "ERROR"